File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    let block = brown + yellow;
    for (let i = Math.ceil(block / 2); i > 1; i--) {
        if (block % i == 0) {
            let divisor = block / i;
            if ((i - 2) * (divisor - 2) === yellow) {
                answer.push(i);
                answer.push(divisor); 
                break; 
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math

def solution(brown, yellow):
    answer = []
    block = brown + yellow
    for i in range(math.ceil(block / 2), 1, -1):
        divisor = block // i
        if (i - 2) * (divisor - 2) == yellow:
            answer.append(i)
            answer.append(divisor)
            break
    return answer

